src/Board.py

```diff
@@ -35,3 +36,243 @@ def __init__(self, style="default", size=5) -> None:
+                        for tile in self.getIndirectlyAdj(
+                            (start[0], start[1] + 1), TerainFilter, shippingLimit - 1
+                        ):
+                            adjacentTileList.append(tile)
+                    else:
+                        adjacentTileList.append((start[0], start[1] + 1))  # bottom left
+            else:  # for odd rows
+                if (
+                    TerainFilter is None
+                    or TerainFilter == self.get(start[0], start[1] + 1).terrain
+                ):
+                    if self.get(start[0], start[1] + 1).terrain == Terrain.RIVER:
+                        for tile in self.getIndirectlyAdj(
+                            (start[0], start[1] + 1), TerainFilter, shippingLimit - 1
+                        ):
+                            adjacentTileList.append(tile)
+                    else:
+                        adjacentTileList.append(
+                            (start[0], start[1] + 1)
+                        )  # bottom right
+                if (
+                    TerainFilter is None
+                    or TerainFilter == self.get(start[0] - 1, start[1] + 1).terrain
+                ):
+                    if self.get(start[0] - 1, start[1] + 1).terrain == Terrain.RIVER:
+                        for tile in self.getIndirectlyAdj(
+                            (start[0] - 1, start[1] + 1),
+                            TerainFilter,
+                            shippingLimit - 1,
+                        ):
+                            adjacentTileList.append(tile)
+                    else:
+                        adjacentTileList.append(
+                            (start[0] - 1, start[1] + 1)
+                        )  # bottom left
+        if start[0] > 0:
+            if (
+                TerainFilter is None
+                or TerainFilter == self.get(start[0] - 1, start[1]).terrain
+            ):
+                if self.get(start[0] - 1, start[1]).terrain == Terrain.RIVER:
+                    for tile in self.getIndirectlyAdj(
+                        (start[0] - 1, start[1]), TerainFilter, shippingLimit - 1
+                    ):
+                        adjacentTileList.append(tile)
+                else:
+                    adjacentTileList.append((start[0] - 1, start[1]))  # left
+        adjacentTileList = self.removeSubsequentDuplicates(adjacentTileList)
+        return adjacentTileList
+
+    def checkAdjacency(self, start, end, shippingLimit) -> str:
+        for tile in self.getDirectlyAdj(start, self.get(end[0], end[1]).terrain):
+            if tile == end:
+                return "Direct"
+        for tile in self.getIndirectlyAdj(
+            start, self.get(end[0], end[1]).terrain, shippingLimit=shippingLimit
+        ):
+            if tile == end:
+                return "Indirect"
+        return "Not"
+
+    def removeSubsequentDuplicates(list=[]) -> list:
```

Comment:
This could be more efficient by using a Set. That should be able to make it O(n) instead of O(n^2). Have a `found` set that you add elements to as well as the `new_list` variable and use the set to check if the item has been found already.